### DISCUSSION TITLE
Does acme.sh remember how I deployed certificates when it renews them?

### DISCUSSION THREAD

**Discussion by ShadowCrafter011** [acmesh-official/acme.sh#4891] [Q&A]
I don't relly know how acme.sh works internally so that's why I'm unsure as to how it'll renew my certificates, thus I have those four questions.

- Does it remember the command I used to deploy the certificates and will it use that again when it renews them? (some env vars set using export are required)
- Also for a certificate to be renewed automatically I just need to issue and deploy it once and make sure I have the cronjob installed right?
- I use DNS to sign a wildcard certificate and for now I always set the API token using an env var. Does renewal work out of the box like this, if not where can I specify the API token?
- If I have a certificate created by another instance of amce.sh how can I also make that it'll get renewed automatically?

Thanks for your answers!

**Comment by neilpang** [comment]
https://github.com/acmesh-official/acme.sh/wiki/dnsapi
